plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java

```diff
@@ -0,0 +1,128 @@
+                    node = new AltibasePlanNode(dataSource, id++, depth, plan, prevNode.getParentNodeAtDepth(depth));
+                }
+            } else { 
+                throw new IllegalStateException("Plan parsing error [depth: " + depth + "]: " + plan + "\n" + planStr);
+            }
+
+            prevNode = node;
+        }
+
+        return rootNodes;
+    }
+
+    /**
+     * Test code
+     */
+    public static void main(String[] args) {
+        String plan = "-----------------------------------------------------------" + "\n"
+                + "PROJECT ( COLUMN_COUNT: 2, TUPLE_SIZE: 34, COST: 151146.46 )" + "\n"
+                + " JOIN ( METHOD: NL, COST: 148444.31 )" + "\n"
+                + "  SCAN ( TABLE: CUSTOMERS, FULL SCAN, ACCESS: ??, COST: 116.76 )" + "\n"
+                + "  VIEW ( ORDERS_T, ACCESS: ??, COST: 14.49 )" + "\n"
+                + "   PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 2.81 )" + "\n"
+                + "    VIEW ( ACCESS: ??, COST: 2.02 )" + "\n"
+                + "     BAG-UNION" + "\n"
+                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
+                + "       SCAN ( TABLE: ORDERS ORDERS_01, INDEX: SYS.ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 )" + "\n"
+                + "        [ VARIABLE KEY ]" + "\n"
+                + "        OR" + "\n"
+                + "         AND" + "\n"
+                + "          CUSTOMERS.CNO = ORDERS_01.CNO" + "\n"
+                + "        [ FILTER ]" + "\n"
+                + "        ORDERS_01.QTY >= 10000" + "\n"
+                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
+                + "       SCAN ( TABLE: ORDERS ORDERS_02, INDEX: SYS.ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 )" + "\n"
+                + "        [ VARIABLE KEY ]" + "\n"
+                + "        OR" + "\n"
+                + "         AND" + "\n"
+                + "          CUSTOMERS.CNO = ORDERS_02.CNO" + "\n"
+                + "        [ FILTER ]" + "\n"
+                + "        ORDERS_02.QTY >= 10000" + "\n"
+                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
+                + "       SCAN ( TABLE: ORDERS ORDERS_03, INDEX: ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 ) " + "\n"
+                + "        [ VARIABLE KEY ]" + "\n"
+                + "        OR" + "\n"
+                + "         AND" + "\n"
+                + "          CUSTOMERS.CNO = ORDERS_03.CNO" + "\n"
+                + "        [ FILTER ]" + "\n"
+                + "        ORDERS_03.QTY >= 10000" + "\n"
+                + "-----------------------------------------------------------"
+                + "* SIMPLE QUERY PLAN";
+
+        try {
+            List<AltibasePlanNode> rootNodes  = AltibasePlanBuilder.build(null, plan);
+
+            System.out.println(rootNodes.get(0).toString4Debug());
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
You can use logging here. log.debug("Message).

Reply:
What about log.debug for line 123 and log.warn for line 125?

Reply:
That sounds ok to me.

Reply:
Thanks for checking.